tests/api.c

```diff
@@ -4540,6 +4544,15 @@ static THREAD_RETURN WOLFSSL_THREAD test_server_nofail(void* args)
         WOLFSSL_METHOD* method = NULL;
         if (cbf != NULL && cbf->method != NULL) {
             method = cbf->method();
+
+#ifdef WOLFSSL_DTLS
+#ifndef NO_OLD_TLS
+            doUdp = cbf->method == wolfDTLS_server_method;
```

Comment:
It's ugly detecting udp by matching the method, but I didn't want to move the `tcp_accept()` after `wolfSSL_new()` because of the comment `/* do it here to detect failure */`, but I can't still get why we can't do accept after `wolfSSL_new()`

Reply:
yes! Fixed as you suggested.